Sources/Services/AWSACM/ACMClient.swift

```diff
@@ -26,30 +26,96 @@ public class ACMClient {
-        let config = try ACMClient.ACMClientConfiguration(region: region)
+    public convenience init(region: Swift.String) async throws {
+        let config = try await ACMClient.ACMClientConfiguration(region: region)
         self.init(config: config)
     }
 
-    public convenience init() async throws {
+    public convenience required init() async throws {
         let config = try await ACMClient.ACMClientConfiguration()
         self.init(config: config)
     }
 }
 
 extension ACMClient {
-    public typealias ACMClientConfiguration = AWSClientConfiguration<ServiceSpecificConfiguration>
+    public class ACMClientConfiguration: AWSDefaultClientConfiguration & AWSRegionClientConfiguration & DefaultClientConfiguration & DefaultHttpClientConfiguration {
+        public var useFIPS: Swift.Bool?
 
-    public struct ServiceSpecificConfiguration: AWSServiceSpecificConfiguration {
-        public typealias AWSServiceEndpointResolver = EndpointResolver
+        public var useDualStack: Swift.Bool?
+
+        public var appID: Swift.String?
+
+        public var credentialsProvider: AWSClientRuntime.CredentialsProviding
+
+        public var awsRetryMode: AWSClientRuntime.AWSRetryMode
+
+        public var region: Swift.String?
+
+        public var signingRegion: Swift.String?
 
-        public var serviceName: String { "ACM" }
-        public var clientName: String { "ACMClient" }
         public var endpointResolver: EndpointResolver
 
-        public init(endpointResolver: EndpointResolver? = nil) throws {
-            self.endpointResolver = try endpointResolver ?? DefaultEndpointResolver()
+        public var encoder: ClientRuntime.RequestEncoder?
+
+        public var decoder: ClientRuntime.ResponseDecoder?
+
+        public var logger: ClientRuntime.LogAgent
+
+        public var retryStrategyOptions: ClientRuntime.RetryStrategyOptions
+
+        public var clientLogMode: ClientRuntime.ClientLogMode
+
+        public var endpoint: Swift.String?
+
+        public var idempotencyTokenGenerator: ClientRuntime.IdempotencyTokenGenerator
+
+        public var httpClientEngine: ClientRuntime.HTTPClient
+
+        public var httpClientConfiguration: ClientRuntime.HttpClientConfiguration
+
+        private init(_ useFIPS: Swift.Bool?, _ useDualStack: Swift.Bool?, _ appID: Swift.String?, _ credentialsProvider: AWSClientRuntime.CredentialsProviding, _ awsRetryMode: AWSClientRuntime.AWSRetryMode, _ region: Swift.String?, _ signingRegion: Swift.String?, _ endpointResolver: EndpointResolver, _ encoder: ClientRuntime.RequestEncoder?, _ decoder: ClientRuntime.ResponseDecoder?, _ logger: ClientRuntime.LogAgent, _ retryStrategyOptions: ClientRuntime.RetryStrategyOptions, _ clientLogMode: ClientRuntime.ClientLogMode, _ endpoint: Swift.String?, _ idempotencyTokenGenerator: ClientRuntime.IdempotencyTokenGenerator, _ httpClientEngine: ClientRuntime.HTTPClient, _ httpClientConfiguration: ClientRuntime.HttpClientConfiguration) {
```

Comment:
nit: is there a way to format this better, or is this convention?

I saw the other comment on the constructor, but don't have an alternative.

Reply:
I can format them into seperate lines to make the `init` look prettier, though I don't think it really matters from user's perspective.

Reply:
Actually it gets more complicated if I want to generate the following:
```
private init(
  _ useFIPS: Swift.Bool?, 
  _ useDualStack: Swift.Bool?, 
  _ appID: Swift.String?, 
  ...
) {
  // 
}
```
I have to open and join two blocks `writer.openBlock(...)`. I would prefer to keep all the constructor args in a single line since it doesn't affect the end user.